submissions/YuraZagor/Frogger/app.js

```diff
@@ -0,0 +1,92 @@
+const allEnemies = [];
+const allSpeeds = [200, 250, 350];
+const rangeY = [73, 156, 239];
+const cellX = 101;
+const cellY = 83;
+const playerStartX = 202;
+const playerStartY = 405;
+const fieldWidth = 505;
+
+function Player(x, y, sprite) {
+    this.x = x;
+    this.y = y;
+    this.sprite=sprite;
+};
+
+Player.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y) 
+};
+
+Player.prototype.update = function() {
+};
+
+const player = new Player ( playerStartX, playerStartY, 'images/char-boy.png' );
+
+function Enemy(x, y, sprite, speedX) {
+    Player.call(this, x, y, sprite);
+    this.speedX = speedX;
+};
+
+Enemy.prototype = Object.create(Player.prototype);
+
+Enemy.prototype.update = function(dt){
+    this.x += this.speedX * dt;
+    if (this.x > fieldWidth){
+        this.x = -55;
+        this.speedX = allSpeeds[Math.floor(Math.random()*3)];
+    };
+    if (this.y === player.y){
+        if ((this.x - player.x < 40 ) && (player.x - this.x < 40 )) {
+            player.x = playerStartX;
+            player.y = playerStartY;
+        };    
+    };
+};
+
+function addBug() {
+    rangeY.forEach ( (levelMark)=>  allEnemies
+    .push( new Enemy(0, levelMark, 'images/enemy-bug.png', allSpeeds[Math.floor(Math.random()*3)] )) 
+    )
+};
+
+player.handleInput = function (direction) {
```

Comment:
One of the purposes of OOP is that class defines all behaviors of the class.
Please explain why this approach was taken.

Reply:
> this kind of behavior is typical only to player object
That's correct. Also true is that any method on `Player` prototype is typical to player object.

`player.propertyName = function ()` worked, as expected. We still can have functions as object properties.

But what if we have a multiplayer game? OOP helps to have objects blue-prints and avoid code duplication.

Oh, I see. You based enemies on `Player`.
So in fact `Player` is an "any" character.

The good thing was that you came up with having a base class for shared properties and behaviour.
The improvemnt would be to have both `Enemy` and `Player` class extend such base class with their own specifics.
Are you up to improve your code accordingly?

Reply:
yes, that way it makes more sense and now I was able to solve ' player.x = playerStartX' omitting .

 Done